src/call-service.js

```diff
@@ -89,7 +89,22 @@ const callService = ({
       throw new Error(`Service ${namespace || defaultNamespace()}.${service}.${handler} not found`);
     }
 
-    const [serviceToRun] = Instances;
+    // Lambda currently searches based on a field in Attributes
+    // called handler, which is the same as InstanceId,
+    // but services which aren't lambdas, use InstanceId.
+    // In future `InstanceId` should refer to the handler name... this is a fallback.
+    let serviceToRun = Instances.find(({ Id }) => Id === handler);
+
+    // If we don't find an instance by instance id, find the first in the list.
+    // Which will likely be a lambda with an incorrect InstanceId. In which case,
+    // it will contain an array with a single item in it.
+    if (!serviceToRun) {
+      [serviceToRun] = Instances;
```

Comment:
Default to current, incorrect behaviour if none found. Phase this out eventually.